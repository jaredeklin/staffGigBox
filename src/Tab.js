import React from 'react'

export const Tab = ({ isActive, tabName, tabIndex, handleTabClick }) => {

  return (
      <li className="tab">
        <a className={`tab-link ${ isActive ? 'active' : '' }`}
           onClick={ () => handleTabClick(tabIndex) }>{ tabName }
        </a>
      </li>
    );
}