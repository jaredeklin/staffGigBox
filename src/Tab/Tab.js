import React from 'react';

export const Tab = ({ isActive, tabName, handleTabClick }) => (
  <li
    className={`tab ${isActive ? 'active' : ''}`}
    onClick={() => handleTabClick(tabName)}
  >
    { tabName }
  </li>
);
