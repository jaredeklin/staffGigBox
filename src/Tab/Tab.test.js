import React from 'react';
import { shallow } from 'enzyme';
import { Tab } from './Tab';

describe('Tab', () => {
  let wrapper;
  const mockHandleTabClick = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Tab
        handleTabClick={mockHandleTabClick}
        isActive={true}
        tabName="Schedule"
      />
    );
  });

  it('should match the snapshot when isActive is true', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when isActive is false', () => {
    wrapper = shallow(
      <Tab
        handleTabClick={mockHandleTabClick}
        isActive={false}
        tabName="Schedule"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleTabClick with correct params when clicked', () => {
    wrapper.find('li').simulate('click');
    expect(mockHandleTabClick).toHaveBeenCalledWith('Schedule');
  });
});
