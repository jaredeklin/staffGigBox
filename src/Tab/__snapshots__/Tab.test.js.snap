// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should match the snapshot when isActive is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    handleTabClick={[Function]}
    isActive={false}
    tabName="Schedule"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Schedule",
      "className": "tab ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Schedule",
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Schedule",
        "className": "tab ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Schedule",
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Tab should match the snapshot when isActive is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    handleTabClick={[Function]}
    isActive={true}
    tabName="Schedule"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Schedule",
      "className": "tab active",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Schedule",
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Schedule",
        "className": "tab active",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Schedule",
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
